collector-kinesis/src/main/java/zipkin/collector/kinesis/KinesisCollector.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.kinesis;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessorFactory;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.KinesisClientLibConfiguration;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker;
+import zipkin.collector.Collector;
+import zipkin.collector.CollectorComponent;
+import zipkin.collector.CollectorMetrics;
+import zipkin.collector.CollectorSampler;
+import zipkin.internal.Util;
+import zipkin.storage.StorageComponent;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.UUID;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+public final class KinesisCollector implements CollectorComponent, Closeable {
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder implements CollectorComponent.Builder {
+
+        Collector.Builder delegate = Collector.builder(KinesisCollector.class);
+
+        AWSCredentialsProvider credentialsProvider;
+        String appName;
+        String streamName;
+        Executor executor = Executors.newSingleThreadExecutor();
```

Comment:
The sqs-collector allows a `parallelism` configuration instead of being able to configure the executor directly. I'm not sure which is better. I guess the other approach more cleanly matches to a property, but that can be handled in the autoconfigure module instead of here